File: tlg/src/main/java/ch/bailu/tlg/StatePaused.java
package ch.bailu.tlg;

public class StatePaused extends State {
    public static final int ID=2;

    public StatePaused(InternalContext c) {
        super(c);
    }

    public State togglePause(PlatformContext c) {
        return new StateRunning(context);
    }

    @Override
    public int getID() {
        return ID;
    }

    @Override
    public String toString() {
        return "Paused";
    }
}

File: tlg/src/main/kotlin/state/StatePaused.kt
package state

import context.InternalContext
import context.PlatformContext


class StatePaused(c: InternalContext) : State(c) {
    override fun togglePause(c: PlatformContext): State {
        return StateRunning(context)
    }

    override val id = ID

    override fun toString(): String {
        return "Paused"
    }

    companion object {
        const val ID = 2
    }
}
